swan-lake/development-tutorials/deployment-guide/k8s-deployment.md

```diff
@@ -0,0 +1,354 @@
+  - /path/to/target/kubernetes/hotel/hotel.ymal
+  - /path/to/target/kubernetes/room/room.yaml
+  - …
+
+patchesStrategicMerge:
+- secret-env-patch.yaml
+
+```
+
+## CI/CD with Kubernetes
+
+This [Ballerina GitHub](https://github.com/ballerina-platform/ballerina-action) action workflow automates the continuous integration and continuous deployment (CI/CD) process for a Ballerina project. It is triggered by every push to the repository and automatically builds the project.
+
+```yaml
+
+name: Ballerina CI/CD with Kubernetes and Ballerina Central Push
+
+# Step 1: Checkout the Repository
+on: [push]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Step 2: Build the Ballerina Project
+      - name: Ballerina Build
+        uses: ballerina-platform/ballerina-action@master
+        with:
+          args: build --cloud=k8s
+
+      # Step 3: Log in to Docker Registry
+      - name: Log in to Docker Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.DOCKER_REGISTRY_URL }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      # Step 4: Push the Docker Image
+      - name: Push Docker Image
+        run: |
+          docker push ${{ secrets.DOCKER_REGISTRY_URL }}/your-docker-repo/your-app:${{ github.sha }}
+
+      # Step 5: Set up Kubectl
+      - name: Set up Kubectl
+        uses: azure/setup-kubectl@v3
+        with:
+          version: 'v1.22.0'
+
+      # Step 6: Deploy to Kubernetes
+      - name: Deploy to Kubernetes
+        env:
+          KUBE_CONFIG_DATA: ${{ secrets.KUBE_CONFIG_DATA }}
+        run: |
+          echo "${KUBE_CONFIG_DATA}" | base64 --decode > $HOME/.kube/config
+          kubectl apply -f <path to generated artifacts>
+```
+The push event can initiate a workflow to the deployment process to the Kubernetes cluster.
+- **Step 02:** This step builds the Ballerina project, compiling the source code and preparing the necessary artifacts for deployment in Kubernetes.
+- **Step 03:** This step logs into a Docker registry (such as Docker Hub or a private registry) to allow pushing the Docker image created in the build to the registry.
+- **Step 04:** Pushes the Docker image to the specified Docker registry.
+- **Step 05:** This step installs kubectl, the Kubernetes command-line tool, which is necessary for managing Kubernetes clusters and deploying applications.
+- **Step 06:** Deploy the application to a Kubernetes cluster by applying the Kubernetes YAML configuration files generated by the Ballerina build in Step 2.
+>**Note:** Replace <path to generated artifacts> with the actual path to the Kubernetes YAML files generated by the Ballerina build command (e.g., target/kubernetes).
+
+See [Code to Cloud deployment](https://ballerina.io/learn/code-to-cloud-deployment/) for more information on deployment options in Kubernetes and Docker.
```

Comment:
Are we going to have https://ballerina.io/learn/code-to-cloud-deployment/ pages as well after these changes?

Reply:
Shouldn't we reword this to `serverless deployment`?